ci/test.sh

```diff
@@ -26,6 +26,31 @@ GTEST_ARGS="--gtest_output=xml:${CUGRAPH_ROOT}/test-results/"
 DOWNLOAD_MODE=""
 EXITCODE=0
 
+PRHTTP=https://api.github.com/repos/rapidsai/cugraph/pulls/${PR_ID}/files
+fnames=`curl -sb -X GET -H "Accept: application/vnd.github.v3+json" -H "Authorization: token $GHTK"  $PRHTTP | python3 -c "import sys, json; print([labels['filename'] for labels in json.load(sys.stdin)])"`
+cpp_changed="false" python_changed="false" nb_changed="false" doc_changed="false" 
+for fname in ${fnames[@]}
+do
+   if [[ "$fname" == *"cpp/cmake/"* || "$fname" == *"cpp/CMakeLists.txt"* || "$fname" == *"cpp/src/"* || "$fname" == *"cpp/include/"* || "$fname" == *"cpp/tests/"* || "$fname" == *"cpp/libcugraph_etl/"* || "$fname" == *"cpp/scripts/"* ]]; then
+      cpp_changed="true" python_changed="true" nb_changed="true" doc_changed="true"
+   fi
+   if [[ "$fname" == *"python/"* ]]; then
+      python_changed="true" nb_changed="true" doc_changed="true"
+   fi
+   if [[ "$fname" == *"docs/"* ]]; then
+      doc_changed="true"
+   fi
+   if [[ "$fname" == *"notebooks/"* ]]; then
+      nb_changed="true"
+   fi
+done
+
+if [[ "$nb_changed" == "false" ]]; then
+   export NB_CHANGED="FALSE"
+else
+   export NB_CHANGED="TRUE"
+fi
+
```

Comment:
I think this code would be more appropriate in `ci/gpu/build.sh` and have the variables exported and then referenced in the lower-level scripts (like you're doing for the `test-notebooks.sh` script).  The main reason being that `test.sh` is sometimes also used by devs outside of CI, and this code would prevent that due to how it's trying to access info about a PR.  It would also mean you wouldn't have to `source test.sh` (which can sometimes have unintended side-effects) since the variables would be set in the top-level calling script.

Once this code is moved out, `test.sh` should be written to also handle cases where the various `python_changed`, `nb_changed`, etc. variables are not set, so devs can continue to run `test.sh` outside of CI.

Reply:
Got it, thanks Rick